### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Lucas-Erkana**
Hi Reviewer

**In this [project](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md), these tasks were completed:**

- [x] Updated UsersController's "show" action to fetch and assign the user and their recent posts. 
- [x] Updated PostsController's "index" action to correctly fetch and assign posts based on the user.
- [x]  Removed unnecessary code from PostsController's "show" action. 
- [x] Added comments for better code readability and understanding.
- [x] Updated the HTML structure to use a <main> element with appropriate id and class attributes.
- [x] Refactor users/index.html.erb view
- [x] Enhance users/show.html.erb view
- [x] Revamp posts/index.html.erb view
- [x] Update posts/show.html.erb view
- [x] Implement the design from the [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#project-requirements).
- [x] Apply styling

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
   end
 
-  def index
-    @users = User.all
+  def show
+    @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+    @comments = Comment.where(post_id: params[:id])
   end
 end
```
Comment (AnselemOdims) [inline app/controllers/users_controller.rb]: - [x] While you have made a great effort in this milestone, you have put the `PostsController` class in the `users_controller.rb` file which might cause some issues as Rails rely on these naming conventions to associate the controller to the correct view file.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+    @user = User.find(params[:id])
+    @posts = @user.recent_post
   end
 
-  def show
-    @post = Post.find_by(user_id: params[:user_id])
+  def index
+    @users = User.all
   end
 end
```
Comment (AnselemOdims) [inline app/controllers/posts_controller.rb]: - [x] While you have made a great effort in this milestone, you have put the `PostsController` class in the `users_controller.rb` file which might cause some issues as Rails rely on these naming conventions to associate the controller to the correct view file.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Lucash2022 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Lucas-Erkana) [reply app/controllers/posts_controller.rb]: @AnselemOdims 
Thank you so much for picking up that error. I have rectified the mistake at this [commit.](https://github.com/Lucash2022/Blog_App/pull/9/commits/d23213d1f2cc919bcdc73fb5aeec3d73fe3d460e)

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (Lucas-Erkana) [reply app/controllers/users_controller.rb]: @AnselemOdims 
Thank you so much for picking up that error. I have rectified the mistake at this [commit.](https://github.com/Lucash2022/Blog_App/pull/9/commits/d23213d1f2cc919bcdc73fb5aeec3d73fe3d460e)

**Review by Owusu-Desmond — APPROVED** [review]
Hi @Lucash2022 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
📌 Linters are passing ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 README file is okay ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**